File: backend/python/chatgpt.py
from openai import OpenAI
from dotenv import load_dotenv

load_dotenv()

client = OpenAI()

def create_letter(cv: str, vac_desc: str) -> str:
    prompt = f"""Я хочу устроиться сотрудником на определенную вакансию.
                Помоги мне составить сопроводительное письмо на основе 
                моего резюме и описания вакансии.Резюме:'{cv}', описание 
                вакансии:'{vac_desc}'."""
    completion = client.chat.completions.create(
        model="gpt-3.5-turbo-0125",
        messages=[
            {"role": "user", "content": prompt}
        ]
    )
    return str(completion.choices[0].message.content)

File: backend/php/src/chatgpt.php
<?php
function askGPT($question){
    $log = require 'logger.php';
    $log->info("Отправляем запрос к ChatGPT с вопросом: " . $question);

    $apiKey = $_ENV["OPENAI_API_KEY"];
    $log->info("Получаем API ключ для ChatGPT");

    $client = OpenAI::client($apiKey);
    $log->info("Создаем клиента OpenAI с переданным ключом");

    try {
        $result = $client->chat()->create([
            'model' => 'gpt-3.5-turbo',
            'messages' => [
                ['role' => 'user', 'content' => $question],
            ],
        ]);
        $log->info("Запрос к ChatGPT успешно выполнен");

        $rating = $result->choices[0]->message->content;
        $log->info("Ответ ChatGPT: " . $rating);
    } catch (Exception $e) {
        $log->error("Ошибка при запросе к ChatGPT: " . $e->getMessage());
        return "Ошибка: не удалось получить ответ от ChatGPT";
    }

    $log->info("Возвращаем полученный ответ от ChatGPT");
    return $rating;
}
?>

